### ISSUE TITLE
First class support for pattern matching CAA queries

### ISSUE THREAD

**Issue by bilal-fazlani** [mkroli/dns4s#11]
```scala
case class QuestionSection(
  qname: String,
  qtype: Int,
  qclass: Int) extends MessageBufferEncoder {
  require(qtype >= 0 && qtype < (1 << 16))
  require(qclass >= 0 && qclass < (1 << 16))
```

qtype of value 257 is CAA 
DNS Certification Authority Authorization, constraining acceptable CAs for a host/domain
RFC 6844
Source: https://en.wikipedia.org/wiki/List_of_DNS_record_types

It would be nice to be able to pattern match and handle these queries. Similar to how I can pattern match on TXT queries
`case Query(q) ~ Questions(QName(host) ~ TypeTXT() :: Nil)`

Context: I am trying to handle CAA queries from let's-encypt
This is the message I received from let's-encrypt

```scala
      Message(
        HeaderSection(61770,false,0,false,false,false,false,0,1,0,0,1),
        Vector(
          QuestionSection(aPPS.BilAL-FazlANi.COM,257,1)
        ),
        Vector(),
        Vector(),
        Vector(ResourceRecord(,41,512,32768,OPTResource(List())))
      )
```

**Comment by bilal-fazlani** [comment]
May be I can do this:

```
case Query(q) ~ Questions(QuestionSection(host, 257, _) :: Nil) => 
```

but I am wondering how can I send a proper CAA response. Like this:

![image](https://user-images.githubusercontent.com/3396271/64115540-8b342f00-cdad-11e9-86ad-801b57f0f8d0.png)

I am guessing creating a new type`CAAResponse` which extends from `trait Resource extends MessageBufferEncoder` would be better.

**Comment by mkroli** [comment]
This is now included in dns4s 0.15.0 thanks to your pull request.
